src/styles/shared/_buttons.scss

```diff
@@ -76,6 +72,9 @@
     border: 1px solid var(--p-border-neutral-subdued);
     border-top-color: var(--p-border-subdued);
     border-bottom-color: var(--p-border-shadow-subdued);
+    transition-property: color, background, border, box-shadow;
+    transition-duration: duration();
+    transition-timing-function: easing();
```

Comment:
Lets remove all transitions. This is not adding any benefit to the user experience right now.

Reply:
We will spend some time looking into component transitions in the future, for now lets remove them.

Reply:
I agree that the `duration` was too slow. 👍🏻 

Also agree with the sentiment that for subtle changes, motion can make them _even more subtle_, therefor lessening the discoverability of the change. However, after speeding up the `duration`, I'm not sure that point is as relevant anymore?

If we are committed to completely removing the button transitions - I'd ask that we consider offering some CSS custom property values that consumers can override. That way, Polaris can default to no transition, but consumers can easily override.

If that is a path we want to pursue - and if there is a non-insignificant time investment - I would then ask if we can defer that to a follow-up PR? Getting this PR merged and available to `online-store-ui` would help us unblock ourselves.

Reply:
I think this PR can remove the transitions, instead of making them consistent. Let me know if you want me to do the work ^

Reply:
I can easily remove the transitions - I was hoping however we could compromise and go the route of CSS Custom Properties so that Polaris can default to no transitions, but I - a consumer - can more easily enable transitions without digging into underlying markup.

Removing transitions altogether invalidates one of my approaches to utilizing Polaris' `Button`:
https://github.com/Shopify/online-store-ui/pull/2143

To be honest I didn't love that solution anyways _(for various reasons)_, its a very gross way of customizing a component, but it did save me from copying much of `Button.tsx` over into my codebase.

Removing transitions forces me into the other approach:
https://github.com/Shopify/online-store-ui/pull/2199

At this point the PR isn't actually providing me any value. Debatable if we want to continue with pushing this forward?

Reply:
Why do the online-store buttons require transitions? Right now I am saying that by default all buttons should not have them?

Reply:
> all buttons should not have them

I wouldn't have expected that opinion to be extended to consumers. I'm left wondering how we determine what styles are open to theme customization while others - such as transitions - are not? How can contributors better understand what fits into theme customization?

I think its fair to say that we have unique concerns for our app. While we conform to the system for much of what we do, we have our own set of problems to solve and therefor our own UI library to achieve that experience.

We utilize motion in much of our UI to enhance the experience. When you remove motion from some of our actions, it begins to feel out of place in respect to the rest of the UI.

I've misinterpreted systems-of-systems, as I wouldn't have expected us to be unable to utilize transitions for our actions. This feels like the wrong decision to me. What led me to this PR was an effort to remove a completely custom UI component in favour of consuming `Polaris.Button` and theming it. I think consumer's will be incentivized to keep building custom components so they can have better control over the desired styles.

When consumers write their own components, where do we look to inform ourselves about how to author certain UI details, such as motion? The currently styleguide doesn't really tell us very much about authoring our own motion styling, and it doesn't seem to include the context you provided from Johan:
https://polaris.shopify.com/design/polaris-is-changing#section-motion

Reply:
I think this is getting a bit into the weeds. We don't want consumers modifying the button transitions yet. We need to do exploration into how buttons should feel and animate.